packages/react/src/components/Dialog/index.tsx

```diff
@@ -178,6 +185,14 @@ const unstable__Dialog = React.forwardRef(
       }
     }, [modal, open]);
 
+    useEffect(() => {
+      if (!open && focusAfterCloseRef) {
+        setTimeout(() => {
+          focusAfterCloseRef.current?.focus();
+        });
```

Comment:
Anytime we step outside of react's scheduling I think it's a good idea to add a comment explaining why.